vendor/processors/stripe_processor.py

```diff
@@ -562,25 +599,28 @@ def get_vendor_offers_in_stripe(self, offer_pk_list, site):
         return offers
 
     def get_vendor_offers_not_in_stripe(self, offer_pk_list, site):
-        offers = Offer.objects.exclude(site=site, pk__in=offer_pk_list)
+        offers = Offer.objects.filter(site=site).exclude(pk__in=offer_pk_list)
         return offers
 
     def create_offers(self, offers):
         for offer in offers:
+            # build product first, since product_id is needed to build price later
             product_data = self.build_product(offer)
-            price_data = self.build_price(offer, offer.current_price_object())
-            coupon_data = self.build_coupon(offer, offer.current_price_object())
 
             new_stripe_product = self.stripe_create_object(self.stripe.Product, product_data)
-            new_stripe_price = self.stripe_create_object(self.stripe.Price, price_data)
-            new_stripe_coupon = self.stripe_create_object(self.stripe.Coupon, coupon_data)
 
             if new_stripe_product:
                 if offer.meta.get('stripe'):
                     offer.meta['stripe']['product_id'] = new_stripe_product['id']
                 else:
                     offer.meta['stripe'] = {'product_id': new_stripe_product['id']}
 
+            price_data = self.build_price(offer, offer.current_price())
+            coupon_data = self.build_coupon(offer, offer.current_price())
```

Comment:
You are always getting a number and never a price instance then why have the check?


If you want to keep the check first try to see if you can retrieve a price instance, to pass it as an argument in the sync. If there is no price instance you can use the `current_price` function